File: NumberGuessingGame.cs
using System;

class NumberGuessingGame
{
    static void Main()
    {
        PlayGame();
    }

    static void PlayGame()
    {
        int targetNumber = GenerateRandomNumber();
        int guessCount = 0;
        bool isGuessedCorrectly = false;

        while (!isGuessedCorrectly)
        {
            string input = GetUserInput();
            
            if (!IsValidGuess(input))
            {
                Console.WriteLine("Invalid input. Please enter a number between 1 and 100.");
                continue;
            }

            int guess = int.Parse(input);
            guessCount++;

            isGuessedCorrectly = CheckGuess(guess, targetNumber, guessCount);
        }
    }

    static int GenerateRandomNumber()
    {
        Random random = new Random();
        return random.Next(1, 101);
    }

    static string GetUserInput()
    {
        Console.Write("Guess a number between 1 and 100: ");
        return Console.ReadLine();
    }

    static bool IsValidGuess(string input)
    {
        return int.TryParse(input, out int number) && number >= 1 && number <= 100;
    }

    static bool CheckGuess(int guess, int targetNumber, int guessCount)
    {
        if (guess < targetNumber)
        {
            Console.WriteLine("Too low. Guess again.");
            return false;
        }
        else if (guess > targetNumber)
        {
            Console.WriteLine("Too high. Guess again.");
            return false;
        }
        else
        {
            Console.WriteLine($"You guessed it in {guessCount} guesses!");
            return true;
        }
    }
}

File: number_guessing_game.py
import random


def is_valid_guess(user_input):
    return user_input.isdigit() and 1 <= int(user_input) <= 100


def main():
    target_number = random.randint(1, 100)
    has_guessed_correctly = False
    guess_count = 0

    user_input = input("Guess a number between 1 and 100: ")

    while not has_guessed_correctly:
        if not is_valid_guess(user_input):
            user_input = input(
                "Invalid input. Please enter a number between 1 and 100: "
            )
            continue

        guess_count += 1
        guessed_number = int(user_input)

        if guessed_number < target_number:
            user_input = input("Too low. Guess again: ")
        elif guessed_number > target_number:
            user_input = input("Too high. Guess again: ")
        else:
            print(f"You guessed it in {guess_count} guesses!")
            has_guessed_correctly = True


if __name__ == "__main__":
    main()
